### PR TITLE
Fix EM issue

### PR DESCRIPTION
**Pull request by tgeral68**
Fix the issue in Expectation Maximization:
- Frechet Mean issue on one element
- Adding projection in Poincaré Ball manifold
- EM issue with Gaussian PDF initialization

### CODE REVIEW

File: geomstats/learning/expectation_maximization.py [hunk 1]
```diff
@@ -312,6 +318,8 @@ def fit(self,
                 label_mask = gs.where(labeled_data[:, 0] == label)
                 grouped_by_label = labeled_data[label_mask][:, 1:]
                 v = variance(grouped_by_label, centroid, self.metric)
+                if grouped_by_label.shape[0] == 1:
```
Comment (clharris) [inline geomstats/learning/expectation_maximization.py:321]: Good catch

**Comment by clharris** [comment]
> Very nice, thanks! Only the github conflicts that need to be solved. @clharris @Hzaatiti do you want to have a look?

Looks good 👍

File: geomstats/geometry/poincare_ball.py [hunk 2]
```diff
@@ -173,6 +173,36 @@ def weighted_gmm_pdf(mixture_coefficients,
 
         return weighted_pdf
 
+    @geomstats.vectorization.decorator(['else', 'vector'])
+    def projection(self, point):
```
Comment (Hzaatiti) [inline geomstats/geometry/poincare_ball.py:177]: Could we have a unit test for this function ?

**Comment by Hzaatiti** [comment]
This is great and should solve the discussed issue #791 !

**Comment by ninamiolane** [comment]
Agreed with @Hzaatiti , adding a unit-test would be great, otherwise we are decreasing code coverage (even for numpy).

**Comment by ninamiolane** [comment]
@tgeral68 update on this?

**Pull request merged**